tests/proc_macro/different_fn_types.rs

```diff
@@ -76,11 +98,13 @@ async fn test_result() {
 async fn get_user_response(uri: &str, role: &str) -> ServiceResponse {
     let mut app = test::init_service(
         App::new()
+            .app_data(web::Data::new(User { id: 1 }))
```

Comment:
How about the pass argument for `Data::new` ? 🤔 
This method is used by other tests, in the future, it can cause implicit consequences when changed.

Reply:
We might get the user simply using a post request with json instead of get request? Removing the need to store the data in the app and using just for the test handler